### PR TITLE
KAFKA-4564: Add system test for pre-0.10 brokers

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
https://jenkins.confluent.io/job/system-test-kafka-branch-builder-2/255/

**Comment by mjsax** [comment]
Call for review @enothereska @dguy

**Review by dguy — APPROVED** [review]
I'm not really sure why we need this, but LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java [hunk 1]
```diff
@@ -305,8 +305,21 @@ public MetadataResponse fetchMetadata() {
      * @throws StreamsException if brokers have version 0.10.0.x
      */
     public void checkBrokerCompatibility() throws StreamsException {
+        final String broker;
+        try {
+            broker = getAnyReadyBrokerId();
+        } catch (final StreamsException e) {
+            if ("Could not find any available broker.".equals(e.getMessage())) {
```
Comment (ijuma) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java]: Relying on the exception message seems fragile.

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java]: I am open to suggestions... Don't have a better idea. :(

Comment (enothereska) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java]: Agree in principle, we should look to remove test reliance on strings, lots of tests have this problem. However, not sure what solution is.

**Comment by enothereska** [comment]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java (hunk 1 above)
Comment (ijuma) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java]: This is not a test though. Can we not throw a more specific exception?

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java]: We can introduce a new class `BrokerNotFoundException extends StreamsException`.

Comment (ijuma) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java]: Seems reasonable

**Comment by mjsax** [comment]
Updated this.

**Comment by mjsax** [comment]
Call for final review and merging @guozhangwang

**Comment by guozhangwang** [comment]
retest this please

**Review by guozhangwang — APPROVED** [review]
Merged to trunk.

**Pull request closed without merging**